packages/overlay/src/integrations/sentry/hooks/useDebounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+
+export default function useDebounce<T>(value: T, delay: number): T {
```

Comment:
This is definitely _NOT_ proper debouncing. It just delays setting the value. A real debounce would not change anything until the last update + debounce period. It also looks like this code was lifted from https://github.com/uidotdev/usehooks/blob/90fbbb4cc085e74e50c36a62a5759a40c62bb98e/index.js#L239-L253. If that's the case, we should give a reference back there.

Finally, if you want to have proper debouncing this answer seems like the way: https://stackoverflow.com/a/75556586/90297. The one below that also shows "the hook way" with useMemo and proper cancellation as it apparently uses the `debounce` decorater from lodash.

Reply:
Hi @BYK 


> https://github.com/uidotdev/usehooks/blob/90fbbb4cc085e74e50c36a62a5759a40c62bb98e/index.js#L239-L253. If that's the case, we should give a reference back there.

No, not lifted from here.



>Finally, if you want to have proper debouncing this answer seems like the way: https://stackoverflow.com/a/75556586/90297. The one below that also shows "the hook way" with useMemo and proper cancellation as it apparently uses the debounce decorater from lodash.

Sure, let me enhance the hook based on the example above. 
But since we are debouncing search. here we will have to maintain 2 states - one for instantly updating the search query in input(this cannot be debounced) and another for passing the search query in children components(this one will be debounced).

Reply:
>But since we are debouncing search. here we will have to maintain 2 states - one for instantly updating the search query in input(this cannot be debounced) and another for passing the search query in children components(this one will be debounced).

Not against debouncing, on the contrary I like it. It's just the debouncing logic is not right here.

Reply:
Yes, even i found that an issue, because we were setting the state again and again in hook, which was not an efficient solution and that would cost us multiple re-renders.